backend/backend_cavity.py

```diff
@@ -119,14 +119,14 @@ def create_faults(self):
             )
 
     def get_fault_counts(
-        self, start_time: datetime, end_time: datetime
+            self, start_time: datetime, end_time: datetime
     ) -> Dict[str, FaultCounter]:
-        result: Dict[str, FaultCounter] = {}
+        result: Dict[str, FaultCounter] = defaultdict(FaultCounter)
 
         for fault in self.faults.values():
-            result[fault.pv.pvname] = fault.get_fault_count_over_time_range(
+            result[fault.tlc] = max(result[fault.tlc], fault.get_fault_count_over_time_range(
```

Comment:
It took me a minute to remember why we did this. It was for the faults that have the same TLC right? Maybe add a comment explaining

Reply:
Yeah! We've got a couple duplicate TLCs that show up multiple times in the spreadsheet (i.e. the fault has two rows in the fault spreadsheet for warning and alarm separately, or there's multiple rows in the spreadsheet because there are multiple PVs associated with a single fault). Attempting to add comment now!